src/cockpit/channels/dbus.py

```diff
@@ -205,7 +240,19 @@ def do_open(self, options):
             if err.errno != errno.EBUSY:
                 raise
 
-        self.ready()
+        if self.name is not None:
+            async def get_ready():
+                async with self.watch_processing_lock:
+                    await self.setup_name_owner_tracking()
+                    if self.owner:
+                        self.ready(unique_name=self.owner)
+                    else:
+                        self.close(problem="not-found")
+            task = asyncio.create_task(get_ready())
+            self.tasks.add(task)
+            task.add_done_callback(self.tasks.discard)
```

Comment:
...at some point i almost start to prefer the explicit callback style...

It would be cool if Python had a syntax that let you use await from non-async functions with the meaning of "turn this into a task now, returning from the original function immediately"...

Short of that I think I might prefer if you handled the `self.name is None` case as an early-exit and unintended this code, providing a large amount of commenting about what's going on here.